x-pack/platform/plugins/shared/streams/server/lib/significant_events/generate_significant_events.ts

```diff
@@ -38,77 +39,80 @@ interface Dependencies {
   logger: Logger;
 }
 
-export async function generateSignificantEventDefinitions(
+export function generateSignificantEventDefinitions(
   params: Params,
   dependencies: Dependencies
-): Promise<GeneratedSignificantEventQuery[]> {
-  const {
-    name,
-    connectorId,
-    currentDate = new Date(),
-    shortLookback = DEFAULT_SHORT_LOOKBACK,
-    longLookback = DEFAULT_LONG_LOOKBACK,
-  } = params;
-  const { inferenceClient, esClient, logger } = dependencies;
-
-  const mend = moment(currentDate);
-  const mstart = mend.clone().subtract(shortLookback);
-
-  const start = mstart.valueOf();
-  const end = mend.valueOf();
-
-  const analysis = await describeDataset({
-    esClient: esClient.client,
-    start,
-    end,
-    index: name,
-  });
-
-  const short = sortAndTruncateAnalyzedFields(analysis);
-
-  const textFields = analysis.fields
-    .filter((field) => field.types.includes('text'))
-    .map((field) => field.name);
-
-  const categorizationField = textFields.includes('message')
-    ? 'message'
-    : textFields.includes('body.text')
-    ? 'body.text'
-    : undefined;
-
-  const lookbackStart = mend.clone().subtract(longLookback).valueOf();
-
-  const logPatterns = categorizationField
-    ? await getLogPatterns({
-        start: lookbackStart,
-        end,
-        esClient,
-        fields: [categorizationField],
-        index: name,
-        includeChanges: true,
-        metadata: [],
-      }).then((results) => {
-        return results.map((result) => {
-          const { sample, count, regex } = result;
-          return { count, sample, regex };
+): Observable<GeneratedSignificantEventQuery[]> {
+  return new Observable<GeneratedSignificantEventQuery[]>((subscriber) => {
```

Comment:
hm I don't think we need to do this right? we just need to open the stream directly in the route to prevent it from timing out. or am I missing something?

Reply:
resolved in https://github.com/elastic/kibana/pull/227671/commits/5453c1954ad1604c4d00b671f46665ca349d3f66